src/templates/specimen/body/annotate/AnnotateSection.js

```diff
@@ -0,0 +1,228 @@
+import { useState, useEffect } from 'react';
+import { Row, Col } from 'react-bootstrap';
+import UserService from 'keycloak/Keycloak';
+import './annotate.css';
+
+/* Import API */
+import GetAnnotations from 'api/annotate/GetAnnotations';
+import InsertAnnotation from 'api/annotate/InsertAnnotation';
+import PatchAnnotation from 'api/annotate/PatchAnnotation';
+import DeleteAnnotation from 'api/annotate/DeleteAnnotation';
+
+/* Import Components */
+import AnnotateRow from './AnnotateRow';
+import AnnotateModal from './AnnotateModal';
+
+
+const AnnotateSection = (props) => {
+    const specimen = props.specimen;
+    const token = UserService.getToken();
+
+    /* Modal handling */
+    const [modalToggle, setModalToggle] = useState(false);
+    const [modalProperty, setModalProperty] = useState({ 'property': '' });
+    const [modalAnnotations, setModalAnnotations] = useState();
+
+    useEffect(() => {
+        SetAnnotations();
+    }, []);
+
+    function SetAnnotations() {
+        /* Search for annotations data and put into view */
+        GetAnnotations(specimen['Meta']['id']['value'], Process);
+
+        function Process(annotations) {
+            const annotationsForModal = {};
+
+            for (const i in annotations) {
+                const annotation = annotations[i];
+
+                if (!annotationsForModal[annotation['target']['indvProp']]) {
+                    annotationsForModal[annotation['target']['indvProp']] = { [annotation['id']]: annotation };
+                } else {
+                    annotationsForModal[annotation['target']['indvProp']][annotation['id']] = annotation;
+                }
+            }
+
+            setModalAnnotations(annotationsForModal);
+        }
+    }
+
+    function ToggleModal(property = null, displayName = null, currentValue = null) {
+        setModalToggle(!modalToggle);
+
+        if (property && property !== modalProperty['property']) {
+            let copyModalProperty = { ...modalProperty };
+
+            copyModalProperty['property'] = property;
+            copyModalProperty['displayName'] = displayName;
+            copyModalProperty['currentValue'] = currentValue;
+
+            setModalProperty(copyModalProperty);
+        }
+    }
+
+    function SaveAnnotation(annotation) {
+        console.log(annotation);
```

Comment:
Misschien deze console log eruit halen voor netheid

Reply:
File is entirely removed in new version